sdk/identity/Azure.Identity/src/DefaultAzureCredentialFactory.cs

```diff
@@ -120,17 +120,23 @@ public virtual TokenCredential CreateWorkloadIdentityCredential()
 
         public virtual TokenCredential CreateManagedIdentityCredential()
         {
-            return new ManagedIdentityCredential(new ManagedIdentityClient(
-                new ManagedIdentityClientOptions
-                {
-                    ResourceIdentifier = Options.ManagedIdentityResourceId,
-                    ClientId = Options.ManagedIdentityClientId,
-                    Pipeline = Pipeline,
-                    Options = Options,
-                    InitialImdsConnectionTimeout = TimeSpan.FromSeconds(1),
-                    ExcludeTokenExchangeManagedIdentitySource = Options.ExcludeWorkloadIdentityCredential
-                })
-            );
+            var options = Options.Clone<DefaultAzureCredentialOptions>();
+            // Set the custom retry policy
+            options.Retry.MaxRetries = 4;
```

Comment:
I would have thought this would be 5?
Other languages set it to that value.

1 attempt + 5 retries

I am assuming the first attempt doesn't count as part of the "retry" count.

Reply:
that's correct - the probe it is not counted. Should it still be 5?

Reply:
I agree the probe request shouldn't count (because it isn't a valid request and we know it will fail deterministically).

> Should it still be 5?

After the probe, I expect their to be 1 "first" attempt (say at 0s) and then 5 retries after 0.8s, 1.6s, 3.2s, 6.4s, 12.8s. To get that behavior, woudl this need to be 5? If so, then yes.